src/signals.jl

```diff
@@ -138,16 +67,92 @@ function _validate_signal_channel(x)
     return x
 end
 
-extract_samples_info(signal) = @compat SamplesInfo(; signal.kind, signal.channels, signal.sample_unit,
-                                                   signal.sample_resolution_in_unit, signal.sample_offset_in_unit,
-                                                   signal.sample_type, signal.sample_rate)
+#####
+##### `onda.samples-info`
+#####
+
+@schema "onda.samples-info" SamplesInfo
+
+@version SamplesInfoV2 begin
+    sensor_type::String
+    channels::Vector{String}
+    sample_unit::String
+    sample_resolution_in_unit::Float64
+    sample_offset_in_unit::Float64
+    sample_type::String = onda_sample_type_from_julia_type(sample_type)
+    sample_rate::Float64
+end
+
+Legolas.accepted_field_type(::SamplesInfoV2SchemaVersion, ::Type{String}) = AbstractString
+Legolas.accepted_field_type(::SamplesInfoV2SchemaVersion, ::Type{Vector{String}}) = AbstractVector{<:AbstractString}
```

Comment:
I should understand this but in the likely case I don't, does this mean:
1. If a table has been serialized with some other `AbstractString` for e.g. `sample_unit`, it can be deserialized OK?
2. do we need `sample_unit::String = string(sample_unit)` in the `@version` invocation? or `String(sample_unit)`? If not, why not?

Reply:
> If a table has been serialized with some other AbstractString for e.g. sample_unit, it can be deserialized OK?

yup 

> do we need sample_unit::String = string(sample_unit) in the `@version` invocation? or String(sample_unit)? If not, why not?

Declaring a required field as `f::T` is Legolas shorthand for `f::T = f`, which is Julia shorthand for `f::T = convert(T, f)`. This matches the previous behavior on `main` which had the `convert` call explicitly (and unnecessarily)

We could add `= string(sample_unit)` to auto-stringify non-string-convertable inputs, but in this case, I'd rather callers just do that themselves if that's what they want